Source/Helpers/MessageProcessor.swift

```diff
@@ -138,15 +128,29 @@ class MessageProcessor {
         return decrypted
     }
 
+    private static func decrypt(_ data: Data, from channel: Storage.Channel) throws -> Data {
+        if channel.type == .singleRatchet {
+            guard let ratchetChannel = try Virgil.ethree.getRatchetChannel(with: channel.getCard()) else {
+                throw UserFriendlyError.noUserOnDevice
+            }
+
+            do {
+                return try ratchetChannel.decrypt(data: data)
+            } catch {
+                return try Virgil.ethree.authDecrypt(data: data, from: channel.getCard())
+            }
```

Comment:
@Ogerets Ugly solution needed to handle a case two Double Ratchet channels are created near the same time to be able to decrypt a Message.NewMessage message.